controllers/common/secret-utils.go

```diff
@@ -27,6 +27,10 @@ const (
 	NamespaceKey                          = "namespace"
 	StorageClusterNameKey                 = "storage-cluster-name"
 	SecretDataKey                         = "secret-data"
+	SecretOriginKey                       = "secret-origin"
+
+	// rook
+	RookOrigin = "rook"
```

Comment:
why do we need this?

Reply:
In my case, I pushed s3 secret as an internal type. But in the future, we may need to introduce different internal secrets which need different logic to handle in the hub.

I added a subtype called origin, Basically, it refers to a place from where this secret originated from. Using this, for the same secret type like blue or green or internal, we can act differently for a different origin.

Reply:
Ok. Should be ok for now. We'll probably need to re-work the entire secret exchange stuff later on.